apf/apfElement.cc

```diff
@@ -119,4 +119,11 @@ void Element::getNodeData()
   field->getData()->getElementData(entity,nodeData);
 }
 
+void Element::getElementNodeData(NewArray<double>& d)
+{
+  d.allocated() ? d.resize(nen) : d.allocate(nen);
```

Comment:
This function is needed to get the node data of the element. I included these allocations/resizing in here so that I don't have to figure out `nen` for the `apf::NewArray` when this function is called elsewhere. The protected member `nen` of the `apf::Element` class is accessible here easily and can be used to allocate/resize if needed.